operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java

```diff
@@ -0,0 +1,201 @@
+
+        // get the service
+        var service = new KeycloakService(k8sclient, kc);
+        String url = "http://" + service.getName() + "." + namespace + ":" + Constants.KEYCLOAK_SERVICE_PORT;
+
+        Awaitility.await().atMost(5, MINUTES).untilAsserted(() -> {
+            Log.info("Starting curl Pod to test if the realm is available");
+            Log.info("Url: '" + url + "'");
+            String curlOutput = K8sUtils.inClusterCurl(k8sclient, namespace, url);
+            Log.info("Output from curl: '" + curlOutput + "'");
+            assertThat(curlOutput).isEqualTo("200");
+        });
+    }
+
+    // local debug commands:
+    //    export TOKEN=$(curl --data "grant_type=password&client_id=token-test-client&username=test&password=test" http://localhost:8080/realms/token-test/protocol/openid-connect/token | jq -r '.access_token')
+    //
+    //    curl http://localhost:8080/realms/token-test/protocol/openid-connect/userinfo -H "Authorization: bearer $TOKEN"
+    //
+    //    example good answer:
+    //    {"sub":"b660eec6-a93b-46fd-abb2-e9fbdff67a63","email_verified":false,"preferred_username":"test"}
+    //    example error answer:
+    //    {"error":"invalid_request","error_description":"Token not provided"}
+    @Test
+    public void testKeycloakCacheIsConnected() {
+        // given
+        Log.info("Setup");
+        var kc = K8sUtils.getDefaultKeycloakDeployment();
+        var crSelector = k8sclient
+                .resources(Keycloak.class)
+                .inNamespace(kc.getMetadata().getNamespace())
+                .withName(kc.getMetadata().getName());
+        var targetInstances = 3;
+        kc.getSpec().setInstances(targetInstances);
+        k8sclient.resources(Keycloak.class).inNamespace(namespace).createOrReplace(kc);
+        var realm = k8sclient.resources(KeycloakRealmImport.class).inNamespace(namespace).load(getClass().getResourceAsStream("/token-test-realm.yaml"));
+        var realmImportSelector = k8sclient.resources(KeycloakRealmImport.class).inNamespace(namespace).withName("example-token-test-kc");
+        realm.createOrReplace();
+
+        Log.info("Waiting for a stable Keycloak Cluster");
+        Awaitility.await()
+                .atMost(10, MINUTES)
+                .pollDelay(5, SECONDS)
+                .ignoreExceptions()
+                .untilAsserted(() -> {
+                    Log.info("Checking realm import has finished.");
+                    CRAssert.assertKeycloakRealmImportStatusCondition(realmImportSelector.get(), KeycloakRealmImportStatusCondition.DONE, true);
+                    Log.info("Checking Keycloak is stable.");
+                    CRAssert.assertKeycloakStatusCondition(crSelector.get(), KeycloakStatusCondition.READY, true);
+                });
+
+        Log.info("Testing the Keycloak Cluster");
+        Awaitility.await().atMost(5, MINUTES).ignoreExceptions().untilAsserted(() -> {
```

Comment:
I wonder, do we really need Awatility here? We know that KC is ready at this point, why should it fail? And i don't want to hear "just in case". 😉 That way we could set some global re-run policy for the whole testsuite.

Reply:
Because caches propagation is, at best of my understanding/experience, only eventually consistent.

I would be more than happy to remove `Awaitility`, let's give it a try in CI.

Reply:
This CI will give us some data:
https://github.com/andreaTP/keycloak/runs/5396139803?check_suite_focus=true

Reply:
Ok, the answer is that the CI doesn't pass:
https://github.com/andreaTP/keycloak/runs/5396139803?check_suite_focus=true#step:9:15317

I'll keep `Awaitility`

Reply:
👍